src/bin/pgcopydb/schema.c

```diff
@@ -971,12 +971,17 @@ struct FilteringQueries listSourceTablesSQL[] = {
 		"               ("
 		"                  select attnum, atttypid::integer, "
 		"                         format('%I', attname) as attname, "
-		"                         i.indrelid is not null as attisprimary "
+		"                         i.indrelid is not null as attisprimary, "
+		"						  d.deptype = 'i' as attisgenerated "
 		"                    from pg_attribute a "
+				"                    from pg_attribute a "
 		"                         left join pg_index i "
 		"                                on i.indrelid = a.attrelid "
 		"                               and a.attnum = ANY(i.indkey) "
 		"                               and i.indisprimary "
+		"					 	  left join pg_depend d "
+		"					 	  		on d.refobjid = a.attrelid "
+		"					 	  		and d.refobjsubid = a.attnum "
```

Comment:
Please also use `pg_depend.classid = 'pg_class':regclass` as a join qualifier.

Reply:
yep, actually I am still not sure if that's enough. It seemed like there are duplicate entries in the attributes because of the join in the pagila rental schema, probably because there are multiple entries in pg_depend for the same attribute? 

I need to refactor the query more.

Reply:
Yeah I see you're only joining to one side of the dependency entry. You need to join with the other side also (`classid` and `objid` in that case), and for that you might need to grab information from other catalogs.